### PR TITLE
bump min deps

### PR DESCRIPTION
**Pull request by jhamman**
The min versions checks are failing in #6475. This hopefully fixes those failures.

- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: ci/requirements/min-all-deps.yml [hunk 1]
```diff
@@ -8,38 +8,38 @@ dependencies:
...
   - lxml=4.6  # Optional dep of pydap
-  - matplotlib-base=3.4
-  - nc-time-axis=1.3
+  - matplotlib-base=3.5
+  - nc-time-axis=1.4
   # netcdf follows a 1.major.minor[.patch] convention
   # (see https://github.com/Unidata/netcdf4-python/issues/1090)
   # bumping the netCDF4 version is currently blocked by #4491
-  - netcdf4=1.5.3
+  - netcdf4=1.5
```
Comment (headtr1ck) [inline ci/requirements/min-all-deps.yml:35]: I think this is still blocked by the comment above?

**Comment by max-sixty** [comment]
> ERROR: Package is too new: flox=0.5 was published on 2022-05-03 which was 12 months ago (policy is 12 months)

I guess that for "experimental" packages like this, we should have different reqs?

(Also the `which was 12 months ago` seems wrong?)

**Review by headtr1ck — COMMENTED** [review]
I don't think that this policy fails if our packages are outdated, only when we require too modern versions....
If it suddenly starts failing this must have other causes.

But anyway good to update this regularly :)

File: ci/requirements/min-all-deps.yml (hunk 1 above)
Comment (jhamman) [reply ci/requirements/min-all-deps.yml:35]: You are correct. I was experimenting with ways to silence this warning: 

```
Analyzing netcdf4...
WARNING: patch version should not appear in requirements file: netcdf4
```

Comment (jhamman) [reply ci/requirements/min-all-deps.yml]: I looked into this a bit more. This comment was misleading. Now that we have pulled pynio from this test environment, we can bump the netcdf4 version. (1.5.3 was released on Oct 22, 2019).

File: ci/min_deps_check.py [hunk 2]
```diff
@@ -157,9 +157,9 @@ def process_pkg(
...
         delta = relativedelta(datetime.now(), policy_published_actual).normalized()
         n_months = delta.years * 12 + delta.months
-        error(
-            f"Package is too new: {pkg}={req_major}.{req_minor} was "
-            f"published on {versions[req_major, req_minor]:%Y-%m-%d} "
+        warning(
+            f"Package is too new: {pkg}={policy_major}.{policy_minor} was "
+            f"published on {versions[policy_major, policy_minor]:%Y-%m-%d} "
             f"which was {n_months} months ago (policy is {policy_months} months)"
         )
```
Comment (jhamman) [inline ci/min_deps_check.py:164]: @max-sixty - this should fix the date error you noticed.

**Comment by dcherian** [comment]
Thanks Joe!

**Pull request merged**